### DISCUSSION TITLE
AdvantEdge with IoT

### DISCUSSION THREAD

**Discussion by alidy97** [InterDigitalInc/AdvantEDGE#428] [Q&A]
Thank you for developing such a great simulation platform!

I want to ask you if i can use this platform with the IoT edge application.
for example i want to build an IoT application on edge server.
also can i use IoT data protocols like MQTT, CoAP....?
Is there any scenarios that already deployed on AdvantEdge related to IoT?

Thank you.

**Comment by alidy97** [comment]
Hi again!
I'm trying to develop an IoT application on Edge platform using MQTT.
so i need to deploy a MQTT client (Publisher or subscriber) on the EU terminal(internal or external), and the MQTT broker on the cloud and the edge.
so is it possible to do that?
and what exactly the steps or the configuration i need to put in each application?   
I hope if someone knows can help me with the answer?

**Comment by dilallkx** [comment]
Hi @alidy97,

Thank you for reaching out, and apologies for the delayed reply!
The platform supports TCP and UDP protocols. MQTT uses TCP and should therefore be able to run in AdvantEDGE.

To do so you will need to do the following:
- Obtain or create a docker container image with your apps
- Create an AdvantEDGE scenario to deploy:
  - Create the desired network topology
  - Expose services at the cloud or edge
  - Include your container image names in the scenario configuration 
- Create a sandbox and deploy your scenario

Note that you can also run your apps external to AdvantEDGE as described here:
https://interdigitalinc.github.io/AdvantEDGE/docs/overview/features/ext-nodes/

For more details on scenario configuration you can take a look at the AdvantEDGE Docs:
https://interdigitalinc.github.io/AdvantEDGE/docs/usage/gui/

I hope this helps you get started. Please let us know if you have additional questions about the platform.

Best regards!
